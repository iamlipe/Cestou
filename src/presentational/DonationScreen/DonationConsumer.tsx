import React from 'react';
import styled from 'styled-components/native';

export const DonationConsumer = () => {
  return (
    <StyledContainer>
      <StyledText>DonationConsumer</StyledText>
    </StyledContainer>
  );
};

const StyledContainer = styled.View``;
const StyledText = styled.Text``;
