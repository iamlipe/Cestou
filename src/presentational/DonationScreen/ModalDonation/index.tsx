import React, {memo} from 'react';
import styled, {useTheme} from 'styled-components/native';
import {t} from 'i18next';
import {GestureHandlerRootView} from 'react-native-gesture-handler';

import {
  StyledModal,
  StyledBackgroundModal,
  StyledContent,
  StyledContainerButtons as ContainerButtons,
  StyledConfirmButton,
  StyledCancelButton,
  StyledCloseButton,
} from '@/components/Modal';

import Counter from '@/components/Counter';

interface Props {
  name: string;
  control: any;
  error?: string;
  onConfirm: () => any;
  onCancel: () => any;
  onClose: () => any;
}

const ModalDonation = ({
  name,
  control,
  error,
  onConfirm,
  onCancel,
  onClose,
  ...rest
}: Props) => {
  const theme = useTheme();

  return (
    <GestureHandlerRootView>
      <StyledModal animationType="fade" transparent {...rest}>
        <StyledBackgroundModal>
          <StyledContent>
            <StyledCloseButton
              name="close"
              size={20}
              color={theme.colors.GRAY_900}
              backgroundColor={theme.colors.BACKGROUND}
              onPress={onClose}
            />

            <StyledTitle>
              {t('text.componentModalDonation.titleModalDonation')}
            </StyledTitle>
            <StyledSubTitle>
              {t('text.componentModalDonation.subTitleModalDonation')}
            </StyledSubTitle>
            <StyledText>
              {t('text.componentModalDonation.textModalDonation')}
            </StyledText>

            <Counter name={name} control={control} />
            {error && <StyledError>{error}</StyledError>}

            <StyledContainerButtons>
              <StyledCancelButton
                buttonColor="text_only"
                textColor="primary"
                size="small"
                title={t('button.cancel')}
                onPress={onCancel}
              />

              <StyledConfirmButton
                size="small"
                title={t('button.confirm')}
                onPress={onConfirm}
              />
            </StyledContainerButtons>
          </StyledContent>
        </StyledBackgroundModal>
      </StyledModal>
    </GestureHandlerRootView>
  );
};

const StyledText = styled.Text`
  width: 100%;
  font-family: ${({theme}) => theme.fonts.REGULAR_SOURCESANSPRO};
  font-size: ${({theme}) => theme.sizing.SMALLER};
  color: ${({theme}) => theme.colors.GRAY_900};
  text-align: center;
  margin-bottom: 16px;
`;

const StyledTitle = styled(StyledText)`
  font-family: ${({theme}) => theme.fonts.SEMIBOLD_SOURCESANSPRO};
  font-size: ${({theme}) => theme.sizing.SMALL};
  color: ${({theme}) => theme.colors.PRIMARY_800};
`;

const StyledSubTitle = styled(StyledText)`
  font-family: ${({theme}) => theme.fonts.SEMIBOLD_SOURCESANSPRO};
`;

const StyledContainerButtons = styled(ContainerButtons)`
  margin-top: 24px;
`;

const StyledError = styled.Text`
  align-self: center;
  font-family: ${({theme}) => theme.fonts.REGULAR_SOURCESANSPRO};
  font-size: ${({theme}) => theme.sizing.SMALLEST};
  color: ${({theme}) => theme.colors.ERROR_800};
  margin-top: 3px;
`;

export default memo(ModalDonation);
