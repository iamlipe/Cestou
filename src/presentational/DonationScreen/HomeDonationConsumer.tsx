import React, {useState} from 'react';
import styled, {useTheme} from 'styled-components/native';
import Icon from 'react-native-vector-icons/MaterialIcons';
import * as Yup from 'yup';
import {t} from 'i18next';
import {Linking} from 'react-native';
import {useForm} from 'react-hook-form';
import {yupResolver} from '@hookform/resolvers/yup';
import {useNavigation} from '@react-navigation/native';
import {NativeStackNavigationProp} from '@react-navigation/native-stack';
import {DonationConsumerStackParamList} from '@/routes/stacks/DonationConsumerStack';
import {OUR_SITE} from 'react-native-dotenv';

import imgOngOne from '@/assets/images/ong_1.png';
import imgOngTwo from '@/assets/images/ong_2.png';

import Header from '@/components/Header';
import CardOng from '@/components/CardOng';
import ModalDonation from './ModalDonation';
import Button from '@/components/Button';

interface DonationsOng {
  image: string;
  title: string;
}

interface DonationForm {
  coins: number;
}

type NavProps = NativeStackNavigationProp<
  DonationConsumerStackParamList,
  'ConfirmDonations'
>;

const schema = Yup.object().shape({
  coins: Yup.number().min(1, t('error.minCoins')).required(t('error.required')),
});

const dataOngServices: DonationsOng[] = [
  {
    image: imgOngOne,
    title: t('cardOng.one'),
  },
  {
    image: imgOngTwo,
    title: t('cardOng.two'),
  },
];

export const HomeDonationConsumer = () => {
  const [isVisibleModal, setIsVisibleModal] = useState(false);
  const {navigate} = useNavigation<NavProps>();
  const theme = useTheme();

  const {
    control,
    handleSubmit,
    formState: {errors, isSubmitted},
  } = useForm<DonationForm>({
    resolver: yupResolver(schema),
  });

  function handleModal(event: boolean) {
    setIsVisibleModal(event);
  }

  function handleNavigateNext() {
    navigate('ConfirmDonations');
    setIsVisibleModal(false);
  }

  async function onSubmit() {
    handleNavigateNext();
  }

  return (
    <StyledContainerScroll showsVerticalScrollIndicator={false}>
      <Header
        title={t('text.homeDonationConsumer.headerTitle')}
        welcome={false}
      />
      <StyledContent>
        <StyledBoxShowCoins>
          <StyledTextCoin>
            {t('text.homeDonationConsumer.titleCoins')}
          </StyledTextCoin>

          <StyledTextQuantityCoins>0</StyledTextQuantityCoins>
          <StyledIconCoin
            name="monetization-on"
            color={theme.colors.PRIMARY_800}
            size={24}
          />
        </StyledBoxShowCoins>
        <StyledTitle>{t('text.homeDonationConsumer.titleSection')}</StyledTitle>
        <StyledText>{t('text.homeDonationConsumer.infoSection')}</StyledText>

        <StyledRow>
          {dataOngServices.map((ong: DonationsOng, index: number) => (
            <CardOng
              key={index}
              title={ong.title}
              image={ong.image}
              onPress={() => handleModal(true)}
              doar
            />
          ))}
        </StyledRow>

        <StyledContainerInfo style={{elevation: 4}}>
          <StyledTextInfo>{t('text.homeDonationConsumer.info')}</StyledTextInfo>
          <Button
            title={t('button.goToSite')}
            onPress={async () => Linking.openURL(OUR_SITE)}
            size="small"
            buttonColor="transparent"
            textColor="primary"
          />
        </StyledContainerInfo>

        {isVisibleModal && (
          <ModalDonation
            name="coins"
            control={control}
            onConfirm={handleSubmit(onSubmit)}
            onCancel={() => handleModal(false)}
            onClose={() => handleModal(false)}
            error={isSubmitted ? errors.coins?.message : ''}
          />
        )}
      </StyledContent>
    </StyledContainerScroll>
  );
};

export const StyledContainerScroll = styled.ScrollView`
  min-height: 100%;
  background-color: ${({theme}) => theme.colors.BACKGROUND};
`;

export const StyledContent = styled.View`
  width: 90%;
  align-self: center;
  padding: 24px 0;
`;

const StyledBoxShowCoins = styled.View`
  min-height: 80px;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
  border: 1px solid ${({theme}) => theme.colors.PRIMARY_800};
  border-radius: 4px;
  padding: 16px;
  margin-bottom: 16px;
`;

const StyledTextCoin = styled.Text`
  width: 80%;
  font-family: ${({theme}) => theme.fonts.SEMIBOLD_SOURCESANSPRO};
  font-size: ${({theme}) => theme.sizing.SMALL};
  color: ${({theme}) => theme.colors.GRAY_900};
`;

const StyledTextQuantityCoins = styled(StyledTextCoin)`
  width: 10%;
  font-size: ${({theme}) => theme.sizing.LARGEST};
`;

const StyledIconCoin = styled(Icon)`
  width: 10%;
`;

const StyledRow = styled.View`
  margin: 16px 0;
  flex-direction: row;
  justify-content: space-between;
`;

const StyledTitle = styled.Text`
  font-family: ${({theme}) => theme.fonts.SEMIBOLD_SOURCESANSPRO};
  font-size: ${({theme}) => theme.sizing.SMALL};
  color: ${({theme}) => theme.colors.GRAY_900};
  margin-bottom: 8px;
`;

const StyledText = styled.Text`
  font-family: ${({theme}) => theme.fonts.REGULAR_SOURCESANSPRO};
  font-size: ${({theme}) => theme.sizing.SMALLER};
  color: ${({theme}) => theme.colors.GRAY_900};
`;

const StyledContainerInfo = styled.View`
  background-color: ${({theme}) => theme.colors.BACKGROUND};
  border: 1px solid ${({theme}) => theme.colors.GRAY_100};
  border-radius: 4px;
  padding: 16px 8px;
`;

const StyledTextInfo = styled.Text`
  margin-bottom: 16px;
`;
