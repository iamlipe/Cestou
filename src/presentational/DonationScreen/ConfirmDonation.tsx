import React from 'react';
import styled from 'styled-components/native';
import {useNavigation} from '@react-navigation/native';
import {useTranslation} from 'react-i18next';

import Coins from '@/assets/svgs/coins.svg';
import Button from '@/components/Button';

import {StyledContainerScroll, StyledContent} from './HomeDonationConsumer';

import Header from '@/components/Header';

export const ConfirmDonations = () => {
  const {goBack} = useNavigation();
  const {t} = useTranslation();

  return (
    <StyledContainerScroll showsVerticalScrollIndicator={false}>
      <Header title={t('text.confirmDonation.headerTitle')} welcome={false} />
      <StyledContent>
        <StyledIconCoins />
        <StyledTitle>{t('text.confirmDonation.title')}</StyledTitle>
        <StyledText>{t('text.confirmDonation.message')}</StyledText>
        <StyledConfirmButton title={t('button.donateAgain')} onPress={goBack} />
      </StyledContent>
    </StyledContainerScroll>
  );
};

const StyledIconCoins = styled(Coins)`
  align-self: center;
  margin-bottom: 24px;
`;

const StyledText = styled.Text`
  font-family: ${({theme}) => theme.fonts.REGULAR_SOURCESANSPRO};
  font-size: ${({theme}) => theme.sizing.SMALLER};
  color: ${({theme}) => theme.colors.GRAY_900};

  text-align: center;
  margin-bottom: 32px;
`;

const StyledTitle = styled(StyledText)`
  font-family: ${({theme}) => theme.fonts.SEMIBOLD_SOURCESANSPRO};
  font-size: ${({theme}) => theme.sizing.LARGER};
  color: ${({theme}) => theme.colors.PRIMARY_800};
`;

const StyledConfirmButton = styled(Button)``;
