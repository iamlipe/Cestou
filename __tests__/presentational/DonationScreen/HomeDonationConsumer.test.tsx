import React from 'react';

import {
  renderWithThemeProvider,
  matchSnapshotWithProvider,
} from '@/config/functions/Testing';

import {HomeDonationConsumer} from '@/presentational/DonationScreen/HomeDonationConsumer';

describe('HomeDonationConsumer', () => {
  test('should render screen correctly', () => {
    renderWithThemeProvider(<HomeDonationConsumer />);
  });

  test('should matches snapshot', () => {
    matchSnapshotWithProvider(<HomeDonationConsumer />);
  });
});
