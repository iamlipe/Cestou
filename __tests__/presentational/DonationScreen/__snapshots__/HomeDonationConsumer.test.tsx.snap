// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeDonationConsumer should matches snapshot 1`] = `
<RCTScrollView
  showsVerticalScrollIndicator={false}
  style={
    Array [
      Object {
        "backgroundColor": "#FFFFFF",
        "minHeight": "100%",
      },
    ]
  }
>
  <View>
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#6CCD91",
              "flexDirection": "row",
              "height": 64,
              "paddingBottom": 0,
              "paddingLeft": "5%",
              "paddingRight": "5%",
              "paddingTop": 0,
              "width": "100%",
            },
          ]
        }
        type="normal"
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "height": 64,
                "justifyContent": "space-between",
                "width": "100%",
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                },
              ]
            }
          >
            <View
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <Text
                allowFontScaling={false}
                selectable={false}
                style={
                  Array [
                    Object {
                      "color": "#1A1A1A",
                      "fontSize": 24,
                    },
                    undefined,
                    Object {
                      "fontFamily": "Material Icons",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
              >
                
              </Text>
            </View>
            <Text
              style={
                Array [
                  Object {
                    "color": "#1A1A1A",
                    "fontFamily": "SourceSansPro-SemiBold",
                    "fontSize": 47,
                    "marginLeft": 10,
                  },
                ]
              }
            >
              Doações
            </Text>
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "alignSelf": "center",
            "paddingBottom": 24,
            "paddingLeft": 0,
            "paddingRight": 0,
            "paddingTop": 24,
            "width": "90%",
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "borderColor": "#00843F",
              "borderRadius": 4,
              "borderStyle": "solid",
              "borderWidth": 1,
              "flexDirection": "row",
              "justifyContent": "space-between",
              "marginBottom": 16,
              "minHeight": 80,
              "paddingBottom": 16,
              "paddingLeft": 16,
              "paddingRight": 16,
              "paddingTop": 16,
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#1A1A1A",
                "fontFamily": "SourceSansPro-SemiBold",
                "fontSize": 31,
                "width": "80%",
              },
            ]
          }
        >
          Saldo de moedas Horticoins gerados da sua cesta
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "#1A1A1A",
                "fontFamily": "SourceSansPro-SemiBold",
                "fontSize": 47,
                "width": "10%",
              },
            ]
          }
        >
          0
        </Text>
        <Text
          allowFontScaling={false}
          selectable={false}
          style={
            Array [
              Object {
                "color": "#00843F",
                "fontSize": 24,
              },
              Array [
                Object {
                  "width": "10%",
                },
              ],
              Object {
                "fontFamily": "Material Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
      </View>
      <Text
        style={
          Array [
            Object {
              "color": "#1A1A1A",
              "fontFamily": "SourceSansPro-SemiBold",
              "fontSize": 31,
              "marginBottom": 8,
            },
          ]
        }
      >
        Ajude no combate a fome
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "#1A1A1A",
              "fontFamily": "SourceSansPro-Regular",
              "fontSize": 27,
            },
          ]
        }
      >
        Veja abaixo as instituições e projetos cadastrados na nossa rede e ajude-nos a atingir a meta do mês doando suas Horticoins que serão convertidas em alimentos.
      </Text>
      <View
        style={
          Array [
            Object {
              "flexDirection": "row",
              "justifyContent": "space-between",
              "marginBottom": 16,
              "marginLeft": 0,
              "marginRight": 0,
              "marginTop": 16,
            },
          ]
        }
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "opacity": 1,
              "width": "48%",
            }
          }
        >
          <Image
            source={
              Object {
                "testUri": "../../../src/assets/images/ong_1.png",
              }
            }
            style={
              Array [
                Object {
                  "borderRadius": 4,
                  "height": 130,
                  "marginBottom": 5,
                  "width": "100%",
                },
              ]
            }
            testID="card-ong-image"
          />
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "#FBEBE8",
                  "borderRadius": 8,
                  "height": 12,
                  "marginBottom": 8,
                  "marginLeft": 0,
                  "marginRight": 0,
                  "marginTop": 8,
                  "width": "100%",
                },
              ]
            }
          >
            <View
              style={
                Array [
                  Object {
                    "backgroundColor": "#FFB394",
                    "borderRadius": 8,
                    "height": 12,
                    "width": "80%",
                  },
                ]
              }
            />
          </View>
          <Text
            style={
              Array [
                Object {
                  "color": "#1A1A1A",
                  "fontFamily": "SourceSansPro-SemiBold",
                  "fontSize": 27,
                },
              ]
            }
          >
            ONG Comida na mesa
          </Text>
        </View>
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "opacity": 1,
              "width": "48%",
            }
          }
        >
          <Image
            source={
              Object {
                "testUri": "../../../src/assets/images/ong_2.png",
              }
            }
            style={
              Array [
                Object {
                  "borderRadius": 4,
                  "height": 130,
                  "marginBottom": 5,
                  "width": "100%",
                },
              ]
            }
            testID="card-ong-image"
          />
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "#FBEBE8",
                  "borderRadius": 8,
                  "height": 12,
                  "marginBottom": 8,
                  "marginLeft": 0,
                  "marginRight": 0,
                  "marginTop": 8,
                  "width": "100%",
                },
              ]
            }
          >
            <View
              style={
                Array [
                  Object {
                    "backgroundColor": "#FFB394",
                    "borderRadius": 8,
                    "height": 12,
                    "width": "80%",
                  },
                ]
              }
            />
          </View>
          <Text
            style={
              Array [
                Object {
                  "color": "#1A1A1A",
                  "fontFamily": "SourceSansPro-SemiBold",
                  "fontSize": 27,
                },
              ]
            }
          >
            ONG Criança sem fome
          </Text>
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#FFFFFF",
              "borderColor": "#F2F2F8",
              "borderRadius": 4,
              "borderStyle": "solid",
              "borderWidth": 1,
              "paddingBottom": 16,
              "paddingLeft": 8,
              "paddingRight": 8,
              "paddingTop": 16,
            },
            Object {
              "elevation": 4,
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "marginBottom": 16,
              },
            ]
          }
        >
          Para saber mais sobre os projetos e instituições que estamos ajudando, acesse o nosso site.
        </Text>
        <View
          accessibilityState={
            Object {
              "disabled": false,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "alignSelf": "center",
              "backgroundColor": "transparent",
              "borderColor": "#00843F",
              "borderRadius": 4,
              "borderStyle": "solid",
              "borderWidth": 1,
              "flexDirection": "row",
              "height": 38,
              "justifyContent": "center",
              "marginBottom": 10,
              "opacity": 1,
              "width": 90,
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#00843F",
                  "fontFamily": "SourceSansPro-SemiBold",
                  "fontSize": 27,
                },
              ]
            }
            textColor="primary"
          >
            Ir para site
          </Text>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
